manifests/job.pp

```diff
@@ -28,10 +33,18 @@
       jobname => $jobname,
     }
   } else {
-    jenkins::job::present { $title:
-      config  => $config,
-      jobname => $jobname,
-      enabled => $enabled,
+    if $source == '' {
+      jenkins::job::present { $title:
+        config  => $config,
+        jobname => $jobname,
+        enabled => $enabled,
+      }
+    } else {
+      jenkins::job::present { $title:
+        config  => file($source),
+        jobname => $jobname,
+        enabled => $enabled,
+      }
```

Comment:
Instead of duplicating the resource, I would suggest using a selector to set a variable for the `config` param.